packages/snjs/lib/services/settings_service/SettingsCache.ts

```diff
@@ -0,0 +1,78 @@
+import { SettingPayload } from '@standardnotes/auth';
+import { SettingsProvider } from './SettingsProvider';
+
+export class SettingsCache implements SettingsProvider {
+  private lastUpdate = 0;
+  private lastUpdateMap: Map<keyof SettingPayload, number> = new Map();
+  private cache: Partial<SettingPayload> = {};
+
+  constructor(
+    private readonly provider: SettingsProvider,
+    private readonly cacheFreshnessMS: number,
+    private readonly currentTime: () => number
+  ) {}
+
+  private setSingle<Key extends keyof SettingPayload>(
+    name: Key,
+    payload: SettingPayload[Key] | null
+  ) {
+    this.cache[name] = payload ?? undefined;
+    this.lastUpdateMap.set(name, this.currentTime());
+  }
+
+  private async setAll(payload: Partial<SettingPayload>) {
+    this.cache = payload;
+    this.cache = await this.provider.listSettings();
+
+    const now = this.currentTime();
+    this.lastUpdate = now;
+    for (let k of this.lastUpdateMap.keys()) {
```

Comment:
```suggestion
    for (const key of this.lastUpdateMap.keys()) {
```

We try to avoid single letter vars to make code more readable and searchable.